pkg/cmd/update-tls-artifacts/generate-owners/tlsmetadatainterfaces/annotation_requirement.go

```diff
@@ -0,0 +1,56 @@
+package tlsmetadatainterfaces
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/openshift/library-go/pkg/certs/cert-inspection/certgraphapi"
+)
+
+func AnnotationInspectRequirement(o AnnotationRequirement, rawData []*certgraphapi.PKIList) (RequirementResult, error) {
```

Comment:
```suggestion
type annotationRequirement struct{
  requirementName string
  annotationName string
}

var _ Requirement = &annotationRequirement{}

func (o &annotationRequirement) InspectRequirement(rawData []*certgraphapi.PKIList) (tlsmetadatainterfaces.RequirementResult, error) {
```

Rather than bare functions, build this as a struct that contains the details.  Something like

```go

```

Reply:
That'd work, but along with `requriementName` and `annotationName` I also have to pass `generateMarkdownFn` so that each class could implement its own markdown report. I could add it to the interface, but then I won't be able to have a common `InspectRequirement` func. Not sure if there is a better solution